### PR TITLE
Pagination + infinite scroller

### PR DESCRIPTION
**Pull request by Jovialiste82**
Hi guys, just finished implementing pagination + infinite scroller. Can you have a look at my PR ?

### CODE REVIEW

File: client/src/pages/HomePage.js [hunk 1]
```diff
@@ -1,56 +1,89 @@
...
+import AppBarLoggedIn from '../layout/AppBarLoggedIn';
+import { Grid, Typography, Box } from '@material-ui/core';
+import { UserContext } from '../context/user';
+import Mention from '../layout/Mention';
+import MentionList from '../layout/MentionList';
+import { makeStyles } from '@material-ui/core/styles';
+import Spinner from '../layout/Spinner';
+import { getMentions } from '../hooks/getMentions';
+import InfiniteScroll from 'react-infinite-scroller';
+import axios from 'axios';
```
Comment (dproc96) [inline client/src/pages/HomePage.js:11]: Can you confirm prettier settings with your team? I think you have single quotes and others have double quotes set.

File: server/controllers/mentionController.js [hunk 2]
```diff
@@ -9,28 +8,50 @@ const Mention = require("../models/mentionModel");
...
+          } else {
+            return { popularity: -1 };
+          }
+        };
+
+        const results = await Mention.find({
+          $or: [...getPlatformsObject(platformsArray)],
+        }).sort(getSortOption(sortOption));
+
+        return results;
```
Comment (dproc96) [inline server/controllers/mentionController.js]: It is often (but not universally) a best practice to only nest functions that have side effects (calling B inside of A changes a variable scoped to A). In this case you could define these functions outside of the getMentions function (you would need to pass platformArray and sortOptions as args though). In that case it's generally preferred to organize them "top down", that is if A calls B and B calls C and then D, define them like this:
```javascript
const A = () => {}
const B = () => {}
const C = () => {}
const D = () => {}
```
Like I said this isn't a universal best practice as far as I know but is generally a good habit to be in for readable code.

**Review by dproc96 — APPROVED** [review]
This generally looks good to go, just had a couple small comments

File: client/src/pages/HomePage.js (hunk 1 above)
Comment (Jovialiste82) [reply client/src/pages/HomePage.js:11]: I switched back to double quotes, but i'll check with the team later today

File: server/controllers/mentionController.js (hunk 2 above)
Comment (Jovialiste82) [reply server/controllers/mentionController.js]: Thx Dan ! It's good to know that. I just moved the functions outside of getMentions.

**Pull request merged**